spec/requests/addresses_spec.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe 'Addresses API', type: :request do
```

Comment:
I think it will be better to write the tests by describing the real class name 
```ruby 
RSpec.describe Api::AddressController, type: :controller do
#then you can use 
 get :index, { user_id: user.id } 
#instead of get "/api/users/#{user_id}/addresses" 
end
````